cli/cli/helpers/grafloki/kubernetes_grafloki.go

```diff
@@ -0,0 +1,477 @@
+	if err := k8sManager.WaitForPodManagedByDeployment(ctx, lokiDeployment, lokiDeploymentMaxRetires, lokiDeploymentRetryInterval); err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred while waiting for pod managed by Loki deployment '%v' to come online.", lokiDeploymentName)
+	}
+
+	lokiService, err := k8sManager.CreateService(ctx,
+		graflokiNamespace,
+		lokiServiceName,
+		map[string]string{}, // empty labels
+		map[string]string{}, // empty annotations
+		lokiLabels,          // match loki deployment pod labels
+		apiv1.ServiceTypeNodePort,
+		[]apiv1.ServicePort{{
+			Name:        "logs-listening",
+			Port:        lokiNodePort,
+			TargetPort:  intstr.FromInt(lokiPort),
+			Protocol:    apiv1.ProtocolTCP,
+			NodePort:    lokiNodePort,
+			AppProtocol: &httpApplicationProtocol,
+		}})
+	shouldRemoveLokiService := false
+	defer func() {
+		if shouldRemoveLokiService {
+			if err := k8sManager.RemoveService(ctx, lokiService); err != nil {
+				logrus.Warnf("Attempted to remove Loki service after an error occurred creating it but an error occurred removing it.")
+				logrus.Warnf("Manually remove Loki service with name: %v", lokiService.Name)
+			}
+		}
+	}()
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred creating Loki service")
+	}
+	lokiHost := getLokiHost(lokiServiceName, graflokiNamespace, lokiNodePort)
+
+	configMapData := map[string]string{
+		"loki-datasource.yaml": fmt.Sprintf(`apiVersion: 1
+datasources:
+  - name: %v
+    type: loki
+    access: proxy
+    url: %v
+    isDefault: true
+    editable: true`, lokiServiceName, lokiHost),
+	}
+	grafanaConfigMap, err := k8sManager.CreateConfigMap(ctx,
+		graflokiNamespace,
+		grafanaDatasourceConfigMapName,
+		map[string]string{}, // empty labels
+		map[string]string{}, // empty annotations
+		configMapData)
+	shouldRemoveGrafanaConfigMap := true
+	defer func() {
+		if shouldRemoveGrafanaConfigMap {
+			if err := k8sManager.RemoveConfigMap(ctx, graflokiNamespace, grafanaConfigMap); err != nil {
+				logrus.Warnf("Attempted to remove Grafana datasource config map after an error occurred creating it but an error occurred removing it.")
+				logrus.Warnf("Manually remove Grafana datasource config map with name: %v", grafanaConfigMap.Name)
+			}
+		}
+	}()
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred creating Grafana datasource configmap.")
+	}
+
+	grafanaDeployment, err := k8sManager.CreateDeployment(
+		ctx,
+		graflokiNamespace,
+		grafanaDeploymentName,
+		grafanaLabels,
+		map[string]string{}, // empty annotations
+		[]apiv1.Container{}, // no init containers
+		[]apiv1.Container{
+			{
+				Name:  "grafana",
+				Image: grafanaImage,
+				Ports: []apiv1.ContainerPort{
+					{
+						Name:          "",
+						ContainerPort: grafanaPort,
+						HostPort:      0,
+						Protocol:      "",
+						HostIP:        "",
+					},
+				},
+				Env: []apiv1.EnvVar{
+					{
+						Name:      "GF_AUTH_ANONYMOUS_ENABLED",
```

Comment:
aren't these envvars used in the Docker version as well? should they be named consts rather than hardcoded, so you don't have to change them in two places if you did in the future?